shark_turbine/kernel/wave/wave_sim.py

```diff
@@ -184,14 +190,38 @@ def decorator(func: Callable[..., Any]) -> Any:
 
 
 def _read_proxy(
-    memory: "Memory", elements_per_thread: Optional[IndexExpr] = None
+    memory: "Memory",
+    elements_per_thread: Optional[IndexExpr] = None,
+    mapping: Optional[IndexMapping] = None,
+    shape: Optional[tuple[IndexExpr, ...]] = None,
 ) -> "Register":
+    if mapping:
+        assert shape
+        mapping_func = mapping.mapping_func
+        res = torch.zeros(shape)

```

Comment:
So for example here, couldn't the shape be inferred from the shape of memory and the mapping_func ?